storage/store.go

```diff
@@ -1587,20 +1602,21 @@ func (s *Store) processRaft() {
 					panic(err) // TODO(bdarnell)
 				}
 			}
+			s.processRaftMu.Unlock()
```

Comment:
This isn't introducing any potential deadlock that wasn't there already, because there was the same relationship between the goroutine itself and `mu`. I don't know of any way to enforce this in go but I'd love to see some annotations and an analyzer like http://clang.llvm.org/docs/ThreadSafetyAnalysis.html